### ISSUE TITLE
VersionNotFound on compatibility check

### ISSUE THREAD

**Issue by nick-zh** [aiven/karapace#479]
If the subject does not exits we used to get SubjectNotExist (or similar), but with the newest update we get `VersionNotFound` for `compatibility/subjects/my-subject/versions/latest`, is that intended?

**Comment by nick-zh** [comment]
I am assuming it is due to this change: https://github.com/aiven/karapace/pull/450
This newly maps the `SchemasNotFoundException` to the same error code as `VersionNotFoundException` which is `VERSION_NOT_FOUND`. I think for `SchemasNotFoundException` we still should get `SUBJECT_NOT_FOUND` as error code, what do you think @jjaakola-aiven ?

**Comment by jjaakola-aiven** [comment]
@nick-zh Thank you for the report. Just a fast thought on this. If the subject does not exist, nor there is any version, the API should return that given schema is compatible. As there is no previous schema to be incompatible with.

**Comment by nick-zh** [comment]
@jjaakola-aiven it depends on what the user wants to know after the request, so the way we handle it is that we catch that exception and set our check to true, others might want to do different thing.
I would advocate that the more information the better so i can differentiate all the cases:
- is really compatible (the schema i try to compare for compatibility exists and is compatible or not)
- version does not exist
- subject does not exist

**Comment by tvainika** [comment]
It looks that Confluent's Schema Registry returns that given schema is compatible. If one needs to know if a subject or a version in a subject exists, there are other API endpoints for those needs.

**Comment by nick-zh** [comment]
Indeed, so then i think instead of the current `VersionNotFoundException` Karapace should also return the same as confluent schema registry if the subject does not exist to be consistent, what do you think @tvainika ?

**Comment by tvainika** [comment]
> Indeed, so then i think instead of the current `VersionNotFoundException` Karapace should also return the same as confluent schema registry if the subject does not exist to be consistent

Yes, that's the way to do it. Let's see when we can fix this, but should be easy one if anyone wants to take a look. Patches welcome.
